### ISSUE TITLE
factor handling in split.data.table, by>1, flatten=F, keep.by=F

### ISSUE THREAD

**Issue by jangorecki** [Rdatatable/data.table#1954]
http://stackoverflow.com/questions/40999920/data-table-into-nested-list
```r
## install.packages(c("gapminder", "data.table"))
library(gapminder)
library(data.table)
gapminder <- data.table(gapminder)
my_table <- gapminder[, .(mdl = .(lm(lifeExp ~ pop + gdpPercap, 
                                  data = gapminder))), 
                          by = .(country, continent)]
first_list <- split(my_table, my_table$continent)
second_list <- lapply(first_list, function(x){ 
                   split(x[, country := as.character(country)], x$country)
               })
final_list <- sapply(second_list, function(x) sapply(x, function(y) y$mdl))

split(my_table, by=c("continent", "country"), keep.by = FALSE, flatten = FALSE)
#Error in `[.data.table`(x, i = make.levels(x, cols = "country", sorted = FALSE),  : 
#  Logical error. Type of column should have been checked by now
```
after convert to character it process fine.

**Comment by jangorecki** [comment]
Issue on hold till #2300 gets resolved.

**Comment by jangorecki** [comment]
This is already fixed by recent commit 862df78.
